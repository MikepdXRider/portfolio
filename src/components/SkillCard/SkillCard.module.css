:root {
  /* alternatives: hsl(225,7%,22%) | rgb(53,55,61)*/
  --darkslategrey: #35373d;
  /* alternatives: hsl(0,100%,100%) | rgb(255,255,255)*/
  --white: #ffffff;
  /* alternatives: hsl(0,0%,100%) | rgb(0,0,0)*/
  --black: #000000;
  /* alternatives: hsl(90,58%,56%) | rgb(143,209,79)*/
  --accentgreen: #8fd14f;
}

.card {
  background: var(--white);
  border-radius: 10px;
  border: var(--darkslategrey) 3px solid;
  padding: 5px;
  min-width: 240px;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 250px;
  text-align: center;
  box-shadow: 5px 5px 10px 5px var(--black);
}

.title {
  color: var(--black);
  text-decoration: underline;
  text-decoration-color: var(--accentgreen);
  text-decoration-thickness: 2px;
  font-size: 24px;
}

.contentList {
  list-style: none;
}

.contentListItem {
  font-size: 20px;
  margin-top: 5px;
}
