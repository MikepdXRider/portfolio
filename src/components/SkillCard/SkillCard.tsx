import React from 'react';
import { Skill } from '../../interfaces/Skill';

export default function SkillCard({ title, content }: Skill) {
  return (
    <article>
      <h3>{title}</h3>
      <ul>
        {content.map((skill) => (
          <li>{skill}</li>
        ))}
      </ul>
    </article>
  );
}
