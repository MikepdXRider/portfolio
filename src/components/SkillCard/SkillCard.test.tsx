// testing
import { render, screen } from '@testing-library/react';
// components
import SkillCard from './SkillCard';

describe('SkillCard component', () => {
  // declares mocks
  const mockSkillTitle = 'test-skill-title';
  const mockSkillContent = [
    'test-skill-one',
    'test-skill-two',
    'test-skill-three',
  ];

  it('renders skillcard', () => {
    render(<SkillCard title={mockSkillTitle} content={mockSkillContent} />);

    // Grabs elements from rendered DOM
    const titleEl = screen.getByText(mockSkillTitle);
    const skillEls = mockSkillContent.map((mockSkill) =>
      screen.getByText(mockSkill)
    );

    // Checks that elements are in rendered DOM
    expect(titleEl).toBeInTheDocument();
    skillEls.map((skillEl) => expect(skillEl).toBeInTheDocument());
  });

  it('matches the existing snapshot', () => {
    const { container } = render(
      <SkillCard title={mockSkillTitle} content={mockSkillContent} />
    );

    expect(container).toMatchSnapshot();
  });
});
