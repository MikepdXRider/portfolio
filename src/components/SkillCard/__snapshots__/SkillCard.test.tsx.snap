// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillCard component matches the existing snapshot 1`] = `
<div>
  <article>
    <h3>
      test-skill-title
    </h3>
    <ul>
      <li>
        test-skill-one
      </li>
      <li>
        test-skill-two
      </li>
      <li>
        test-skill-three
      </li>
    </ul>
  </article>
</div>
`;
